File: src/main/java/gauge/language/psi/impl/SpecNamedElementImpl.java
package gauge.language.psi.impl;

import com.intellij.extapi.psi.ASTWrapperPsiElement;
import com.intellij.lang.ASTNode;
import com.intellij.psi.PsiElement;
import com.intellij.psi.PsiElementFactory;
import gauge.language.psi.SpecNamedElement;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public abstract class SpecNamedElementImpl extends ASTWrapperPsiElement implements SpecNamedElement {
  public SpecNamedElementImpl(@NotNull ASTNode node) {
    super(node);
  }

  @Nullable
  @Override
  public PsiElement getNameIdentifier() {
    // 名前を示す要素を取得するロジックを実装してください
    // 例: return findChildByType(SPEC_IDENTIFIER);
    return null;
  }

  @Override
  public PsiElement setName(@NotNull String name) {
    PsiElement nameIdentifier = getNameIdentifier();
    if (nameIdentifier != null) {
      // PSIツリーの名前部分を新しい名前で置き換え
      PsiElement newNameIdentifier = PsiElementFactory.getInstance(getProject())
              .createIdentifier(name);
      nameIdentifier.replace(newNameIdentifier);
    }
    return this;
  }
}

File: src/main/kotlin/com/github/glider2355/gaugetsplugin/language/psi/impl/SpecNamedElementImpl.kt
package com.github.glider2355.gaugetsplugin.language.psi.impl

import com.github.glider2355.gaugetsplugin.language.psi.SpecNamedElement
import com.intellij.extapi.psi.ASTWrapperPsiElement
import com.intellij.lang.ASTNode

abstract class SpecNamedElementImpl(node: ASTNode) : ASTWrapperPsiElement(node), SpecNamedElement {
    init {
        requireNotNull(node) { "ASTNode cannot be null" }
    }
}